tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature

```diff
@@ -712,3 +712,39 @@ Feature: Sharing files and folders with internal users
       | shared-resource |
       | lorem.txt       |
       | simple-folder   |
+
+  Scenario: change existing expiration date of an existing share to a date beyond the enforced maximum expiry date
+    Given the setting "shareapi_default_expire_date_user_share" of app "core" has been set to "yes"
+    And the setting "shareapi_expire_after_n_days_user_share" of app "core" has been set to "7"
+    And user "user1" has created a new share with following settings
+      | path       | lorem.txt |
+      | shareWith  | user2     |
+      | expireDate | +7        |
+    And the setting "shareapi_enforce_expire_date_user_share" of app "core" has been set to "yes"
+    And user "user1" has logged in using the webUI
+    When the user tries to edit the collaborator expiry date of "User Two" of resource "lorem.txt" to "+14" days using the webUI
+    Then the expiration date field should be marked as "required" on the WebUI
+    And the expiration date for "+14 days" should be disabled
+    And the expiration date shown on the webUI should be "+7" days
+    And it should not be possible to save the pending share on the webUI
+
+  Scenario: setting default expiration date and enforcing it does not change the expiration date of a previously created share
+    Given user "user1" has created a new share with following settings
+      | path      | lorem.txt |
+      | shareWith | user2     |
+    And the setting "shareapi_default_expire_date_user_share" of app "core" has been set to "yes"
+    And the setting "shareapi_expire_after_n_days_user_share" of app "core" has been set to "7"
+    And the setting "shareapi_enforce_expire_date_user_share" of app "core" has been set to "yes"
```

Comment:
IMO this is a when step as you're checking reaction of `enforced-expiration-setting` !

Reply:
We are checking the reaction of the UI to that setting. We know that the share will not be changed by the setting as we tested that by API tests in core, but here we want to make sure that the UI does not accidentally show the new default expire date when it should not